### PR TITLE
Add CBRT to the V2 engine

### PR DESCRIPTION
**Pull request by margarit-h**
Signed-off-by: Margarit Hakobyan <margarith@bitquilltech.com>

### Description
Adds CBRT() math function to the V2 engine.

Usage: CBRT(number) calculates the cube root of a number
Argument type: INTEGER/LONG/FLOAT/DOUBLE
Return type: DOUBLE

(Non-negative) INTEGER/LONG/FLOAT/DOUBLE -> DOUBLE
(Negative) INTEGER/LONG/FLOAT/DOUBLE -> DOUBLE

Example::

    opensearchsql> SELECT CBRT(8), CBRT(9.261), CBRT(-27);
    fetched rows / total rows = 1/1
    +-----------+---------------+-------------+
    | CBRT(8)   | CBRT(9.261)   | CBRT(-27)   |
    |-----------+---------------+-------------|
    | 2.0       | 2.1           | -3.0        |
    +-----------+---------------+-------------+
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
  - [x] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunction.java [hunk 1]
```diff
@@ -471,6 +472,21 @@ private static DefaultFunctionResolver sqrt() {
...
+   * Definition of cbrt(x) function.
+   * Calculate the cube root of a non-negative number x
+   * The supported signature is
+   * INTEGER/LONG/FLOAT/DOUBLE -> DOUBLE
+   */
+  private static DefaultFunctionResolver cbrt() {
+    return FunctionDSL.define(BuiltinFunctionName.CBRT.getName(),
+        ExprCoreType.numberTypes().stream()
+            .map(type -> FunctionDSL.impl(FunctionDSL.nullMissingHandling(
+                v -> v.doubleValue() < 0 ? ExprNullValue.of() :
```
Comment (Yury-Fridlyand) [inline core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunction.java:485]: Function should accept negative numbers too:
```math
\sqrt[3]{-8}=-2
```

**Review by Yury-Fridlyand — CHANGES_REQUESTED** [review]

File: core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunction.java (hunk 1 above)
Comment (Yury-Fridlyand) [reply core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunction.java:485]:
```math
\sqrt[3]{-8}=-2
```

File: core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java [hunk 2]
```diff
@@ -2327,4 +2327,79 @@ public void tan_missing_value() {
...
+    FunctionExpression cbrt = dsl.cbrt(DSL.literal(value));
+    assertThat(cbrt.valueOf(valueEnv()), allOf(hasType(DOUBLE), hasValue(Math.cbrt(value))));
+    assertEquals(String.format("cbrt(%s)", value), cbrt.toString());
+  }
+
+  /**
+   * Test cbrt with long value.
+   */
+  @ParameterizedTest(name = "cbrt({0})")
+  @ValueSource(longs = {1L, 2L})
```
Comment (acarbonetto) [inline core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java:2346]: we could also test negative longs, and larger-than-2 longs

File: core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java [hunk 3]
```diff
@@ -2327,4 +2327,79 @@ public void tan_missing_value() {
     assertEquals(DOUBLE, tan.type());
     assertTrue(tan.valueOf(valueEnv()).isMissing());
   }
+
+  /**
+   * Test cbrt with int value.
+   */
+  @ParameterizedTest(name = "cbrt({0})")
+  @ValueSource(ints = {1, 2})
```
Comment (acarbonetto) [inline core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java:2335]: negative integers?

File: core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java [hunk 4]
```diff
@@ -2327,4 +2327,79 @@ public void tan_missing_value() {
...
+    FunctionExpression cbrt = dsl.cbrt(DSL.literal(value));
+    assertThat(cbrt.valueOf(valueEnv()), allOf(hasType(DOUBLE), hasValue(Math.cbrt(value))));
+    assertEquals(String.format("cbrt(%s)", value), cbrt.toString());
+  }
+
+  /**
+   * Test cbrt with float value.
+   */
+  @ParameterizedTest(name = "cbrt({0})")
+  @ValueSource(floats = {1F, 2F})
```
Comment (acarbonetto) [inline core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java:2357]: negative floats, or floats with more precision?

File: core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java [hunk 5]
```diff
@@ -2327,4 +2327,79 @@ public void tan_missing_value() {
...
+    FunctionExpression cbrt = dsl.cbrt(DSL.literal(value));
+    assertThat(cbrt.valueOf(valueEnv()), allOf(hasType(DOUBLE), hasValue(Math.cbrt(value))));
+    assertEquals(String.format("cbrt(%s)", value), cbrt.toString());
+  }
+
+  /**
+   * Test cbrt with double value.
+   */
+  @ParameterizedTest(name = "cbrt({0})")
+  @ValueSource(doubles = {1D, 2D})
```
Comment (acarbonetto) [inline core/src/test/java/org/opensearch/sql/expression/operator/arthmetic/MathematicalFunctionTest.java:2368]: larger doubles, doubles with values below the decimal?

**Review by MaxKsyunz — APPROVED** [review]

**Review by GumpacG — APPROVED** [review]

**Review by forestmvey — APPROVED** [review]

**Review by acarbonetto — APPROVED** [review]
